bench: update random value generation

AryanJ18 opened the pull request:
Resolves #4989 

## Description

> What is the purpose of this pull request?

This pull request:

-  It initialized the random number generation out of the benchmarking loops . 
- Replaced the use of randu with uniform

## Related Issues

> Does this pull request have any related issues?

-  NO

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

In the issue it is mentioned to use uniform and discreteUniform to replace randu but I did not find any case in which discreteUniform could be used as per my knowledge

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

AryanJ18 committed:
Refactor random number generation in JS benchmarks for stats/base/dists/triangular #4989

Resolves #4989

AryanJ18 committed:
fixes

chore: fix lint issues and reorder requires

bench: fixes a issue in random number generation

AryanJ18 committed:
chore: fix lint issues and reorder requires

AryanJ18 committed:
chore: fix lint issues

AryanJ18 committed:
chore: fix lint issues

AryanJ18 commented:
Hi @anandkaranubc, I’ve made sure all benchmarks pass locally and I’ve addressed all the lint errors that showed up so far (spacing, import order, header formatting, etc.).
However, I keep running into new lint issues one after another, even after I think I’ve fixed everything. Is there a better workflow or tool I can use to catch all lint errors *before* pushing or opening a PR?
Also, let me know if I should squash these commits or leave them as is.

AryanJ18 committed:
chore: fix lint issue

kgryte commented:
@AryanJ18 Thank you for your PR. However, your proposed changes contain significant deviations from our style conventions. I suppose going back through and comparing to other packages to ensure that everything looks and feels like other packages (e.g., in terms of spacing, etc.).

AryanJ18 committed:
chore: fix lint errors

AryanJ18 committed:
chore: fix lint issue

AryanJ18 committed:
chore: fix lint issue

AryanJ18 committed:
chore:fix lint issue

AryanJ18 committed:
chore: fix lint issue

AryanJ18 committed:
chore: fix lint issue

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/ctor/benchmark/benchmark.js:
```diff
@@ -32,16 +33,27 @@ var Triangular = require( './../lib' );
 
 bench( pkg+'::instantiation', function benchmark( bm ) {
 	var dist;
+	var as = new Float64Array(bm.iterations);
+	var bs = new Float64Array(bm.iterations);
```
This should be changed to follow what is done in https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/base/dists/arcsine/ctor/benchmark/benchmark.js

Like:

```js
	var dist;
    var len;
	var a;
	var b;
	var c;
	var i;

    len = 100;
	a = new Float64Array( len );
	b = new Float64Array( len );
	c = new Float64Array( len );
    for ( i = 0; i < len; i++ ) {
		a[ i ] = uniform( EPS, 10.0 );
		b[ i ] = uniform( a[ i ] + EPS, a[ i ] + 10.0 + EPS );
        c[ i ] = uniform( a[ i ], b[ i ] );
	}
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/ctor/benchmark/benchmark.js:
```diff
@@ -32,16 +33,27 @@ var Triangular = require( './../lib' );
+		a = as[i];
+		b = bs[i];
+		c = cs[i];
 		dist = new Triangular( a, b, c );
```
```suggestion
		dist = new Triangular( a[ i%len ], b[ i%len ], c[ i%len ] );
```

After the above changes have been made.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/ctor/benchmark/benchmark.js:
```diff
@@ -98,7 +110,7 @@ bench( pkg+'::set:a', function benchmark( bm ) {
 	bm.tic();
 	for ( i = 0; i < bm.iterations; i++ ) {
-		y = ( 100.0*randu() ) + EPS;
+		y = uniform(EPS, 100.0);
```
This needs to be moved outside the benchmarking loop, similar to what's being done above.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/ctor/benchmark/benchmark.js:
```diff
@@ -155,7 +167,7 @@ bench( pkg+'::set:b', function benchmark( bm ) {
 	bm.tic();
 	for ( i = 0; i < bm.iterations; i++ ) {
-		y = ( 100.0*randu() ) + c + EPS;
+		y = uniform(EPS+c, 100.0+c+EPS);
```
Same comment and applied throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/entropy/benchmark/benchmark.js:
```diff
@@ -44,9 +44,9 @@ bench( pkg, function benchmark( b ) {
-		max[ i ] = ( randu()*10.0 ) + min[ i ] + EPS;
-		mode[ i ] = ( ( max[ i ] - min[ i ] ) * randu() ) + min[ i ];
+		min[ i ] = uniform( 0.0, 10.0 );
+		max[ i ] = uniform( min[ i ] + EPS, 10.0 + min[ i ] + EPS );
```
```suggestion
		max[ i ] = uniform( min[ i ] + EPS, min[ i ] + 10.0 + EPS );
```

Just to be consistent with the other benchmarks. This change applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/triangular/entropy/benchmark/benchmark.native.js:
```diff
@@ -53,9 +53,9 @@ bench( pkg+'::native', opts, function benchmark( b ) {
-		max[ i ] = ( randu()*10.0 ) + min[ i ] + EPS;
-		mode[ i ] = ( ( max[ i ] - min[ i ] ) * randu() ) + min[ i ];
+		min[ i ] = uniform( 0.0, 10.0 );
+		max[ i ] = uniform( min[ i ] + EPS, 10.0 + min[ i ] + EPS );
```
Same comment as above. Rest looks good :)

anandkaranubc commented:
@AryanJ18 The PR looks in good shape overall. Thank you for making the changes as Athan suggested. Just a few changes remain, as mentioned above.

AryanJ18 commented:
Thank you! I’ll make the remaining changes as suggested and push the update shortly.

AryanJ18 committed:
fix: changes made as suggested by maintainer

-moved random number generation out of loops
-fixed some more identation for more consistency in the code

AryanJ18 committed:
chore : fix lint issues

AryanJ18 committed:
chore: fix lint issues

AryanJ18 committed:
chore: fix lint issues

AryanJ18 commented:
Hi @anandkaranubc , just checking in! I wanted to kindly follow up on this PR. Please let me know if there’s anything else I need to change or if there's any feedback I can work on. Looking forward to your thoughts!
